arangod/Cluster/HeartbeatThread.cpp

```diff
@@ -1306,8 +1306,15 @@ void HeartbeatThread::updateAgentPool(VPackSlice const& agentPool) {
       agentPool.hasKey("size") && agentPool.get("size").getUInt() > 0) {
     try {
       std::vector<std::string> values;
+      // we have to make sure that the leader is on the front
+      auto leaderId = agentPool.get("id").stringRef();
+      auto pool = agentPool.get("pool");
+      values.emplace_back(pool.get(leaderId).copyString());
```

Comment:
We can also add a `values.reserve(pool.length());` here to save some reallocations later.